components/brave_vpn/browser/connection/connection_api_impl.cc

```diff
@@ -172,7 +174,20 @@ void ConnectionAPIImpl::ResetHostname() {
 }
 
 bool ConnectionAPIImpl::SmartRoutingEnabled() const {
-  return hostname_ ? hostname_->smart_routing_enabled : false;
+  if (hostname_ && hostname_->smart_routing_enabled) {
```

Comment:
I manually tested each variation here

Here's how the logs look like for each:
```
SmartRoutingEnabled Host "zurich-ipsec-6.sudosecuritygroup.com" does not support smart proxy routing.
SmartRoutingEnabled Host "chicago-3.sgw.guardianapp.com" supports smart proxy routing. Smart proxy routing will be used.
SmartRoutingEnabled Host "sanjose-3.sgw.guardianapp.com" supports smart proxy routing. However, user has disabled smart proxy routing.
```